sets/ordered_set.go

```diff
@@ -65,16 +69,22 @@ func (s OrderedSet[T]) Clone() OrderedSet[T] {
 	return maps.Clone(s)
 }
 
-// AsSlice returns the set as a sorted slice.
+// Returns the set as a sorted slice.
 func (s OrderedSet[T]) AsSlice() []T {
 	rv := make([]T, 0, len(s))
-	for x, _ := range s {
+	for x := range s {
 		rv = append(rv, x)
 	}
 	slices.Sort(rv)
 	return rv
 }
 
+// Returns the set as a Set. Changes to the returned Set will be reflected in
+// this set.
```

Comment:
I'm not sure I like this.  Adding extra forms of aliasing seems like a bad feature for the library.

I think the motivation is to do equality-checking of ordered sets and non-ordered sets?  And be able to map over either sort? (Or is there another reason?)

I think this is probably the best solution if that's the reason.  (Stupid lack of double-dispatch, and Go not having iterators as a first-class object.)  The Go-ish option is to have an interface which both Set and OrderedSet obey, which contains the stuff necessary for equality-testing, but I'm not enough a purist to insist on that.  (Plus I have no idea if interfaces can have type parameters?)

Reply:
The motivation is that `OrderedSet[T]` ought to be a subtype of `Set[T]`: some functions might take `Set`, and we'd like to be able to use an `OrderedSet` instead. While we could do an explicit cast via `Set[T](orderedSet)`, this feels to me like it breaks abstraction, since the cast is only possible because the underlying representations are the same. But maybe I'm just trying too hard to have nice abstractions in a language like Go!

Reply:
Oh, forgot to mention: interfaces can have type parameters, but their functions can't. Makes it hard to define those first-class iterators ourselves. 😞